sklearn/_fast_gradient_boosting/binning.py

```diff
@@ -0,0 +1,140 @@
+"""
+This module contains the BinMapper class.
+
+BinMapper is used for mapping a real-valued dataset into integer-valued bins.
+Bin thresholds are computed with the quantiles so that each bin contains
+approximately the same number of samples.
+"""
+# Author: Nicolas Hug
+
+import numpy as np
+
+from ..utils import check_random_state, check_array
+from ..base import BaseEstimator, TransformerMixin
+from ..utils.validation import check_is_fitted
+from ._binning import _map_to_bins
+from .types import X_DTYPE, X_BINNED_DTYPE
+
+
+def _find_binning_thresholds(data, max_bins=256, subsample=int(2e5),
```

Comment:
we kinda have the idea that for _private_ functions, it's better not to have default values, it's proven to prevent certain bugs.

Reply:
I didn't know that thanks, I'll update.
Out of curiosity would you have a ref to an issue were it was a problem?

Reply:
Can't remember an issue, but we've refused to set default values on some private functions even when a user raises that as an issue here IIRC.